core/tests/org.openjdk.jmc.jdp.test/src/test/java/org/openjdk/jmc/jdp/client/TestToolkit.java

```diff
@@ -148,4 +150,62 @@ public int getMaxHeartBeatTimeout() {
 		}
 		return bytes;
 	}
+
+	/**
+	 * See https://bugs.openjdk.java.net/browse/JMC-7539
+	 */
+	public static boolean areBroadcastingTestsEnabled() {
+		if (System.getProperty("skipJDPMulticastTests", "false").equals("true")) {
+			JDPClientTest.LOGGER.log(Level.INFO, "Broadcasting related tests are disabled");
+			return false;
+		}
+		if (!isBroadcastingPossible()) {
+			String os = System.getProperty("os.name").toLowerCase();
+			String helpMessage = ", you could try ";
+			String address = TEST_MULTICAST_ADDRESS.getCanonicalHostName();
+			if (os.startsWith("mac") || os.startsWith("darwin")) {
+				helpMessage += String.format("'sudo route add -host %s -interface en0'", address);
+			} else if (os.startsWith("linux")) {
+				helpMessage += String.format("'sudo ip route add %s dev eth0'", address);
+			} else {
+				helpMessage = "";
+			}
+			JDPClientTest.LOGGER.log(Level.WARNING, "Broadcasting does not seem to be possible.\n"
+					+ "This is usually related to VPN. There are four possible ways to remedy this:\n"
```

Comment:
Actually, you enumerate only 3 possibilities, although the last one consists of two distinct options, depending on your build scenario

Reply:
Oh, I thought I changed it.